### ISSUE TITLE
Changes requested in County Update table on SoS page

### ISSUE THREAD

**Issue by sfsinger19103** [FreeAndFair/ColoradoRLA#655]
- Audited contest discrepancies, non-audited contest discrepancies. Plus column for each “reason for audit”.
- A count of the number of ballot cards that have discrepancies in audited contests, and a count of the number of ballot cards that have discrepancies in non-audited contests (noting that some ballot cards will count in both categories)

Per 9/5 conversation with CDOS:
Take out "discrepancy" column.
Add these columns (suggested column titles included, but they might be too long):

- <span>#</span> of ballot cards showing a discrepancy in at least one audited contest "Discrepancies in Contests Under Audit"

For each "reason for audit" (currently "state contest, county contest"), make a column
- <span>#</span> of ballot cards showing a discrepancy in at least one race under audit for the given reason "Discrepancies in State Contests", "Discrepancies in County Contests"

- <span>#</span> of ballot cards showing a discrepancy in at least one non-audited contest "Discrepancies in Contests Not Under Audit"

- <span>#</span> of ballot cards not found "Ballot Cards Not Found"

**Comment by sfsinger19103** [comment]
CDOS has promised (9/5) to sketch the County Update Table that they want and share it with us soon. So there may be more changes in this spec.

**Comment by dmzimmerman** [comment]
There will be no columns for "reasons to audit" without significant rearchitecture of some things, which as far as I can see means it's not happening in the near future. But the rest of this should be addressed soon, server-side.

**Comment by dmzimmerman** [comment]
Which is to say: there will be a column for "discrepancies in audited contests", and a column for "discrepancies in contests not under audit", but not for "audited contests with reason X" and "audited contests with reason Y".

**Comment by dmzimmerman** [comment]
Actually, I want some clarification from them about exactly what they want here. If they want the discrepancies for each category of audited contest (opportunistic contests fall under that list too), there's a good way to do it that feels less "hack"-ish to me than just "let's add yet another field to the dashboard" - namely, replacing the existing "discrepancies" field with a mapping from category to number of discrepancies. But I don't want to get involved in doing that if it's not going to give them what they want.

**Comment by sfsinger19103** [comment]
@dmzimmerman Your immediately previous comment applies to the "Someday" milestone, not the current milestone, right? My current plan is to tell CDOS simply "no" on "audited columns with reason X" and "audited columns for reason Y" for the November 2017 system. Let me know if you want me to change that plan.

If not, are you asking for clarification from them now about what they might want for 2018 and beyond?

**Comment by dmzimmerman** [comment]
This was discussed out of band. Basically, I want to do this once, and do it right, rather than jumping through refactoring and database changes every time somebody wants a new column for some category of audit.

**Comment by sfsinger19103** [comment]
My current understanding is that if CDOS gives us a spec by Friday September 8 for the County Update table involving columns discussed in this thread, we'll be able to implement them for the September 15 deliverable.

**Comment by dmzimmerman** [comment]
The server side of this is done.

**Comment by ranweiler** [comment]
Blocked on CDOS for updated spec.

**Comment by sfsinger19103** [comment]
CDOS 9/11:
The County Updates table should show the total number of discrepancies by audited contests and non-audited contests. The status messages that we have seen so far, or want added, are included in the status column for reference.

Not started	Shown when the state has not started the audit
File upload deadline missed	Shown for counties that did not upload files before the audit started
Ballot manifest uploaded	Shown when counties have successfully hashed and uploaded their ballot manifest
CVR uploaded	Shown when counties have successfully hashed and uploaded their CVR
Ballot manifest & CVR uploaded Shown when counties have successfully hashed and uploaded their ballot manifest and CVR
Ballot Manifest/CVR mismatch	Shown when counties successfully upload a mismatched ballot manifest and CVR
Audit underway 	Shown for counties that uploaded files before the state initiated the audit
Audit complete	Shown when a county has completed all county contest audits, met the risk limit, and there are no state contests
Round complete	Shown when a county has completed the current round of the audit, with discrepancies that will create a second round, or there is a state contest

The total number of discrepancies in each contest could be displayed in the Contest Updates table since that table currently contains a list of contests by county.

See item 4 in the [CDOS word document](https://drive.google.com/open?id=0B5_BzTbq3XxPLUJZQVEwNnhuZEU) for table mock-ups.

**Comment by dmzimmerman** [comment]
```
Audit complete	Shown when a county has completed all county contest audits, met the risk limit, and there are no state contests
Round complete	Shown when a county has completed the current round of the audit, with discrepancies that will create a second round, or there is a state contest
```
Are they still under the impression that the tool will handle state contests?

**Comment by ranweiler** [comment]
Made a separate issue for the per-contest totals: #710
